node/impl/full/eth_utils.go

```diff
@@ -251,7 +251,11 @@ func newEthBlockFromFilecoinTipSet(ctx context.Context, ts *types.TipSet, fullTx
 			return ethtypes.EthBlock{}, xerrors.Errorf("failed to convert msg to ethTx: %w", err)
 		}
 
-		tx.ChainID = ethtypes.EthUint64(build.Eip155ChainId)
+		// Don't override the chainID if this is not an EIP-1559 transaction.
+		// The chainID has already been set to 0 for leagcy ETH transactions.
+		if smsg.Signature.Type != crypto.SigTypeDelegated {
```

Comment:
Im confused by this, won't the Signature.Type be delegated for both Eip155 txs and legacy txs?  Doesn't this code avoid setting the ChainID for Eip155 Txs as well as legacy?

Reply:
Actually we don't need this at all as `ChainID` is already set on tx in all cases by the `newEthTxFromSignedMessage` method.